### PR TITLE
Add ability to define a tree structure in a multimap.

### PR DESCRIPTION
**Pull request by rehans**
Please review and if ok, please merge.

### CODE REVIEW

**Comment by rehans** [comment]
@mgee If time allows, can you please adapt the travis.yml so the build succeeds? I think installing cmake might not be necessary any longer, right?

**Comment by mengelbrecht** [comment]
@rehans I pushed a fix to master. Please rebase onto master, then it should work.

File: include/kompositum/builder.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#include <kompositum/composite.h>
+#include <kompositum/leaf.h>
+#include <kompositum/util.h>
+#include <map>
+
+namespace Kompositum {
+using ParentChildTreeDef = std::multimap<IDType, IDType>;
+
+ComponentPtr buildComposite(IDType id, const ParentChildTreeDef& treeMap) {
+    bool hasChildren = treeMap.find(id) != treeMap.end();
```
Comment (mengelbrecht) [inline include/kompositum/builder.h]: Missing const.

File: test/basic.cpp [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+#include <kompositum/composite.h>
+#include <kompositum/leaf.h>
 #include <kompositum/printer.h>
+#include <kompositum/util.h>
 
 using namespace Kompositum;
 
+static const ParentChildTreeDef kTreeDef = {
+    {1, 2}, {1, 3}, {1, 4}, {4, 5}, {4, 6}, {1, 7},
+
```
Comment (mengelbrecht) [inline test/basic.cpp]: Remove empty line

File: include/kompositum/builder.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+#include <kompositum/visitor.h>
+#include <kompositum/component.h>
+#include <kompositum/composite.h>
+#include <kompositum/leaf.h>
+#include <kompositum/util.h>
+#include <map>
+
+namespace Kompositum {
+using ParentChildTreeDef = std::multimap<IDType, IDType>;
```
Comment (mengelbrecht) [inline include/kompositum/builder.h:15]: Missing newline above typedef.

**Review by mengelbrecht — APPROVED** [review]

**Pull request merged**